include/ap/argument_parser.hpp

```diff
@@ -59,6 +60,113 @@ namespace ap {
 
 class argument_parser;
 
+namespace error {
```

Comment:
Before the namespace add an error base class:

```
class argument_parser_error : public std::runtime_error {
public:
    explicit argument_parser_error(const std::string& msg)
    : std::runtime_error(msg) {}
};
```

All other error classes should derive from this base class